tests/rest/admin/test_user.py

```diff
@@ -1507,41 +1520,89 @@ def test_change_name_deactivate_user_user_directory(self):
 
         self.assertEqual(200, int(channel.result["code"]), msg=channel.result["body"])
         self.assertEqual("@user:test", channel.json_body["name"])
-        self.assertEqual(True, channel.json_body["deactivated"])
+        self.assertTrue(channel.json_body["deactivated"])
         self.assertEqual("Foobar", channel.json_body["displayname"])
 
         # is not in user directory
         profile = self.get_success(self.store.get_user_in_directory(self.other_user))
-        self.assertTrue(profile is None)
+        self.assertIsNone(profile)
 
     def test_reactivate_user(self):
         """
         Test reactivating another user.
         """
 
         # Deactivate the user.
+        self._deactivate_user("@user:test")
+
+        # Attempt to reactivate the user (without a password).
         channel = self.make_request(
             "PUT",
             self.url_other_user,
             access_token=self.admin_user_tok,
-            content=json.dumps({"deactivated": True}).encode(encoding="utf_8"),
+            content=json.dumps({"deactivated": False}).encode(encoding="utf_8"),
```

Comment:
We can pass just the dictionary in here instead of worrying about dumping and encoding (and the framework will handle that). It should simplify some of the code in here a bit! (There's a bunch of places to make this change, I only marked this one.)

```suggestion
            content={"deactivated": False},
```